vendor/processors/stripe.py

```diff
@@ -1,104 +1,261 @@
-
+import logging
 import stripe
-
+from django.conf import settings
 from .base import PaymentProcessorBase
-from vendor.models.choice import PurchaseStatus
+from vendor.models.choice import (
+    SubscriptionStatus,
+    TransactionTypes,
+    PaymentTypes,
+    TermType,
+    TermDetailUnits,
+    InvoiceStatus,
+    PurchaseStatus
+)
+from vendor.integrations import StripeIntegration
+
+logger = logging.getLogger(__name__)
 
 class StripeProcessor(PaymentProcessorBase):
+    """
+    Implementation of Stripe SDK
+    https://stripe.com/docs/api/authentication?lang=python
+    """
 
-    def processor_setup(self):
-        stripe.api_key = settings.STRIPE_TEST_PUBLIC_KEY    # TODO: This should work, but may not the best way to do this
+    TRANSACTION_SUCCESS_MESSAGE = 'message'
+    TRANSACTION_SUCCESS_CODE = 'code'
+    TRANSACTION_FAIL_MESSAGE = 'error_text'
+    TRANSACTION_FAIL_CODE = 'error_code'
+    TRANSACTION_RESPONSE_CODE = 'response_code'
+    transaction_submitted = False
 
-    def get_checkout_context(self, request=request, context={}):
-        '''
-        The Invoice plus any additional values to include in the payment record.
-        '''
-        context = super().get_checkout_context(request=request, context=context)
-        context['integration_check'] = 'accept_a_payment'
+    products_mapping = {}
 
-        metadata = {}
-        metadata['integration_check'] = 'accept_a_payment'
-        metadata['order_id'] = str(self.invoice.pk)
+    def get_checkout_context(self, request=None, context={}):
+        context = super().get_checkout_context(context=context)
+        # TODO need to figure out how we're building stripe form
+        """if 'credit_card_form' not in context:
+            context['credit_card_form'] = CreditCardForm(initial={'payment_type': PaymentTypes.CREDIT_CARD})
+        if 'billing_address_form' not in context:
+            context['billing_address_form'] = BillingAddressForm()"""
+        return context
 
-        intent = stripe.PaymentIntent.create(
-            amount=int(self.invoice.total * 100),  # Amount in pennies so it can be an int() rather than a float
-            currency=self.invoice.currency,        # "usd"
-            metadata=metadata,
-        )
+    def processor_setup(self, site, source=None):
+        self.credentials = StripeIntegration(site)
+        self.stripe_source = source
+        self.site = site
+        if self.credentials.instance:
+            stripe.api_key = self.credentials.instance.private_key
+        elif settings.STRIPE_API_KEY:
+            stripe.api_key = settings.STRIPE_PUBLIC_KEY
+        else:
+            logger.error("StripeProcessor missing keys in settings: STRIPE_PUBLIC_KEY")
+            raise ValueError("StripeProcessor missing keys in settings: STRIPE_PUBLIC_KEY")
 
-        context['client_secret'] = intent.client_secret
-        context['pub_key'] = settings.STRIPE_TEST_PUBLIC_KEY
+        self.initialize_products()
```

Comment:
I don't think that this goes here. I would remove it for the time being. We can talk about when and were should you initialize the products